pkg/ruler/api_test.go

```diff
@@ -296,18 +374,27 @@ func TestRuler_PrometheusRules(t *testing.T) {
 
 			a := NewAPI(r, r.store, log.NewNopLogger())
 
-			req := requestFor(t, http.MethodGet, "https://localhost:8080/prometheus/api/v1/rules", nil, userID)
+			req := requestFor(t, http.MethodGet, "https://localhost:8080/prometheus/api/v1/rules"+tc.queryParams, nil, userID)
 			w := httptest.NewRecorder()
 			a.PrometheusRules(w, req)
 
 			resp := w.Result()
 			body, _ := io.ReadAll(resp.Body)
+			if tc.expectedStatusCode != 0 {
+				require.Equal(t, tc.expectedStatusCode, resp.StatusCode)
+			} else {
+				require.Equal(t, http.StatusOK, resp.StatusCode)
+			}
 
-			// Check status code and status response
 			responseJSON := response{}
 			err := json.Unmarshal(body, &responseJSON)
 			require.NoError(t, err)
-			require.Equal(t, http.StatusOK, resp.StatusCode)
+
+			if tc.expectedErrorType != "" {
+				assert.Equal(t, responseJSON.Status, "error")
```

Comment:
[nit] Expected value should be first parameter
```suggestion
				assert.Equal(t, "error", responseJSON.Status)
```